src/js/node/http.ts

```diff
@@ -1328,8 +1320,8 @@ export class ClientRequest extends OutgoingMessage {
       } else {
         protocol = defaultAgent.protocol || "http:";
       }
-      this.#protocol = protocol;
     }
+    this.#protocol = protocol;
```

Comment:
Can you verify that redirects from https -> http and https -> http still work? I don’t know if we have a test for that.